apis/compute/v1alpha1/substitutions_types.go

```diff
@@ -0,0 +1,14 @@
+package v1alpha1
+
+// Substitution defines the substitution configuration
+type Substitution struct {
+	// +kubebuilder:validation:Required
+	// +immutable
+	// +kubebuilder:validation:Enum=ipv4Address;ipv6Address
+	Type string `json:"type" yaml:"type"`
+	// +kubebuilder:validation:Required
+	// +immutable
+	Key                  string            `json:"key" yaml:"key"`
+	Unique               bool              `json:"unique" yaml:"unique"`
```

Comment:
we should mark this also as immutable to prevent -> false -> duplicates -> true states